File: multi level/hrrn.c
int pidd(int t,int * a,int *b, int n)
{
	int f=0,maxrr=-999,i,x,min,index;
	for(i=0;i<n;i++)
	{
		if(a[i]<=t)
		{
			f=1;
			x=((t-a[i])+b[i])/b[i];
			if(x>maxrr)
			{
				maxrr=x;
				index=i;
			}
		}
	}
	if(f==0){
		min=999;
	for(i=0;i<n;i++)
	{
		if(min>a[i])
		{
			min=a[i];
			index=i;
	   }
	}}
	return index;
}
int hrrn(int t,int *a,int *b,int n)
{
	int j=0,p;
	while(j<n)
	{
		p=pidd(t,a,b,n);
		if(a[p]<=t)
		{
			t+=b[p];
			a[p]=999;
			printf(" p%d %d|",p+1,t);
			j++;
		}
		else
		{
			t=a[p];
			printf(" /// %d |",t);
		}
	}return t;
}

File: python/hrrn.py
a=list(map(int,input("enter arrival times").split()))
b=list(map(int,input("enter burst times").split()))
i=t=0;c=[0,0,0,0,0,0,0,0];d=a[:]
while i<len(a):
    p=a.index(min(a))
    if(a[p]<=t):
        m=[[],[]]
        for k in range(len(a)):
            if(a[k]<=t):
                m[0].append(k)
                m[1].append(((t-a[k])+b[k])/b[k])
        p=m[0][m[1].index(max(m[1]))]
        t=t+b[p]
        a[p]=99;
        print (" p%d %d|"%(p+1,t),end ="")
        i=i+1
        c[p]=t;
    else:
        t=a[p]
        print(" /// %d|"%t,end="")
print("\nct\ttat\twt")
for i in range(len(a)):
    print("%d\t%d\t%d\t"%(c[i],c[i]-d[i],c[i]-d[i]-b[i]))
    
        
    
